semeio/jobs/correlated_observations_scaling/job_config.py

```diff
@@ -84,76 +84,123 @@ def _min_max_value(value):
+"""
+
+_KEYS_SCHEMA = {
+    MK.ElementValidators: (_is_not_empty,),
+    MK.Type: types.List,
+    MK.Content: {
+        MK.Item: {
+            MK.Type: types.NamedDict,
+            MK.Content: {
+                "key": {
+                    MK.Type: types.String,
+                    MK.Description: "Name of the key. An asterisk is accepted"
+                    " as a suffix to expand all matching keywords (e.g."
+                    " WOPR* will include WOPR:OP1)",
+                },
+                "index": {
+                    MK.Type: types.List,
+                    MK.LayerTransformation: _to_int_list,
+                    MK.Description: "Indexes matching the data points relevant"
+                    " for update. Accepts single integer and ranges e.g."
+                    " (1,2,4-6,14-15) ->[1, 2, 4, 5, 6, 14, 15]",
+                    MK.Content: {
+                        MK.Item: {
+                            MK.Type: types.Integer,
+                            MK.ElementValidators: (_min_value,),
+                        }
+                    },
+                },
+            },
+        }
+    },
+}
+
+
 def build_schema():
     return {
         MK.Type: types.NamedDict,
         MK.Description: "Keys and index lists from all scaled keys",
         MK.LayerTransformation: _expand_input,
         MK.Content: {
             "CALCULATE_KEYS": {
-                MK.Required: True,
                 MK.Type: types.NamedDict,
+                MK.Description: _CALCULATE_KEYS_DESCRIPTION,
+                MK.ElementValidators: (_is_not_empty,),
                 MK.Content: {
-                    "keys": {
-                        MK.Required: True,
-                        MK.Type: types.List,
-                        MK.Content: {
-                            MK.Item: {
-                                MK.Type: types.NamedDict,
-                                MK.Content: {
-                                    "key": {MK.Required: True, MK.Type: types.String},
-                                    "index": {
-                                        MK.Required: False,
-                                        MK.Type: types.List,
-                                        MK.LayerTransformation: _to_int_list,
-                                        MK.Content: {
-                                            MK.Item: {
-                                                MK.Type: types.Integer,
-                                                MK.ElementValidators: (_min_value,),
-                                            }
-                                        },
-                                    },
-                                },
-                            }
-                        },
-                    },
+                    "keys": _KEYS_SCHEMA,
                     "threshold": {
-                        MK.Required: False,
+                        MK.AllowNone: True,
```

Comment:
Does it make sense to allow the `threshold` to be `None`? Should we not instead provide a default value? I think there already is a layer that currently keeps the default values...

Reply:
I actually did provide a Default when I added the description in https://github.com/equinor/semeio/pull/124, but as `MK.Default` was not accessible for `0.5.3` I removed it again, just for the sake of that PR. I'll definitively add it instead - and then we don't need `MK.AllowNone` of course.